src/compiler/types/validate-build-package-json.ts

```diff
@@ -171,25 +259,53 @@ export const validateCollection = (
   }
 };
 
+/**
+ * Check that the `browser` field is set correctly in `package.json` for the
+ * `DIST_COLLECTION` output target.
+ *
+ * @param config the stencil config
+ * @param compilerCtx the current compiler context
+ * @param buildCtx the current build context
+ */
 export const validateBrowser = (config: d.Config, compilerCtx: d.CompilerCtx, buildCtx: d.BuildCtx) => {
   if (isString(buildCtx.packageJson.browser)) {
     const msg = `package.json "browser" property is set to "${buildCtx.packageJson.browser}". However, for maximum compatibility with all bundlers it's recommended to not set the "browser" property and instead ensure both "module" and "main" properties are set.`;
     packageJsonWarn(config, compilerCtx, buildCtx, msg, `"browser"`);
   }
 };
 
+/**
+ * Build a package.json error message with a special header and so on.
+ *
+ * @param config the stencil config
+ * @param compilerCtx the current compiler context
+ * @param buildCtx the current build context
+ * @param msg an error string
+ * @param warnKey a warning key
```

Comment:
it might be worth expounding on what these variables do a little bit.  without reading the code, it's hard to figure out what it means to be 'a warning key'

Reply:
it's not a great parameter name either, basically the field is supposed to be the field which caused the error, so the original line number can be found in the source JSON file. I'm going to rename it to `jsonField`